src/Interpreters/ActionsDAG.cpp

```diff
@@ -1943,6 +1943,10 @@ ActionsDAGPtr ActionsDAG::cloneActionsForFilterPushDown(
     }
 
     auto conjunction = getConjunctionNodes(predicate, allowed_nodes);
+    if (conjunction.allowed.size() == 1 && conjunction.rejected.size() == 1
```

Comment:
I forgot to ask why this condition `conjunction.allowed.size() == 1` is required?